### PR TITLE
Docs improvements

### PR DESCRIPTION
**Pull request by ppetrosh**

### CODE REVIEW

**Comment by ppetrosh** [comment]
@smittytone thank you for the changes. Can you please resolve the conflicts so that we can merge the branch?

**Comment by alexey-nbl** [comment]
misprints in the main readme: onConneced, thee

**Comment by ragrus-nbl** [comment]
@smittytone Please apply our corrections (notes on Service Accounts and etc.) while resolving the conflicts - they are important

File: README.md [hunk 1]
```diff
@@ -279,247 +287,288 @@ client.enableCfgReceiving(onConfigReceived, onDone);
...
-- If no device found, the method tries to register the new one.
-- Else if a device is found and the keys are identical, the method succeeds, assuming the device is already registered.
-- Otherwise, it is assumed that **another** device is registered with the same ID and the method returns the `GOOGLE_IOT_CORE_ERROR_ALREADY_REGISTERED` error.
+This method registers the device in Google IoT Core. It performs a minimal registration: only one private-public key pair, without expiration, is registered.
 
-**If you are going to use this method, add** `#require "OAuth2.agent.lib.nut:2.0.0"` **to the top of your agent code**.
+The method attempts to see if there is a device with the same ID as the one specified in the client’s constructor. If it finds a match, it compares the device's public key with the supplied key. If the keys match, the methods succeeds &mdash; it is assumed that the specified device has already been registered. Otherwise, it is assumed that *another* device is registered with the same ID, and the method returns the *GOOGLE_IOT_CORE_ERROR_ALREADY_REGISTERED* error.
```
Comment (ragrus-nbl) [inline README.md]: `the methods succeeds` -> `the method succeeds`

File: README.md [hunk 2]
```diff
@@ -279,247 +287,288 @@ client.enableCfgReceiving(onConfigReceived, onDone);
...
-| *state* | String or Blob  | Yes | [Device state](https://cloud.google.com/iot/docs/concepts/devices#device_state). Application specific data. Application can use the [Serializer](https://developer.electricimp.com/libraries/utilities/serializer) library to convert Squirrel objects to Blobs. |
-| *onReported* | Function  | Optional | [Callback](#callback-onreportedstate-error) called when the operation is completed or an error occurs. |
+| *state* | String or blob | Yes | A [device state](https://cloud.google.com/iot/docs/concepts/devices#device_state) record. The application can use the [Serializer](https://developer.electricimp.com/libraries/utilities/serializer) library to convert Squirrel objects to blobs |
+| *onReported* | Function  | Optional | A [callback](#onreported) executed when the operation is completed or an error occurs |
 
-##### Callback: onReported(*state, error*) #####
+#### onReported ####
 
-This callback is called when the state is considered as reported or an error occurs.
+The *onReported* parameter takes a function that has two parameter of its own:
```
Comment (ragrus-nbl) [inline README.md]: `two parameter` -> `two parameters`

**Review by smittytone — APPROVED** [review]

**Comment by ragrus-nbl** [comment]
@smittytone As far as I can see, you have declined the corrections from `develop` branch while resolving the conflicts. Please recover that corrections.

**Comment by ppetrosh** [comment]
I will do now. We have to merge the branch ASAP.

**Pull request merged**